src/app/Processor/ProcessorDetailPage/ProcessorDetailPage.tsx

```diff
@@ -119,11 +126,17 @@ const ProcessorDetailPage = (): JSX.Element => {
 
   useEffect(() => {
     setCurrentProcessor(processor);
+    if (processor && !canEditResource(processor?.status)) {
+      setProcessorRefreshInterval(refreshInterval);
+    } else {
+      setProcessorRefreshInterval(0);
```

Comment:
IDK if I misunderstand.. 

when you want to clear the polling should you not also call the `clearInterval()` [method](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/main/src/hooks/usePolling/usePolling.ts#L20) returned from `usePolling(..)`? Or, should the call to `usePolling(..)` have a dependency on `refreshInterval` to ensure it is set to `0`?

Reply:
The way to stop `usePolling` is to pass it a delay of `0`.

https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/b919ee59566392bad320630c250238ffdd693bae/src/hooks/usePolling/usePolling.ts#L13-L21

When doing so, its `useEffect` will run again because it depends on this delay value. 

Before the new render, `useEffect` will run its cleanup function, the one it's being returned inside it, clearing the existing `setInterval`.

On the re-render, a new `setInterval` will not be set because the delay is `0`.

(You can find more about this mechanism [here](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1), especially in the last paragraph)

I'm [passing](https://github.com/kelvah/sandbox-ui/blob/fbbf0d706ec42e88a54467aa6ad2fce0ab23cd21/src/app/Processor/ProcessorDetailPage/ProcessorDetailPage.tsx#L107-L110) `processorRefreshInterval` to `usePolling`, and setting It to `0` or to `refeshInterval` when needed.
`refreshInterval` is only a constant to store the delay value in a single place.

Going back to your original question, `usePolling` does not return anything! You may ask why then its return type is `void | (() => void)`. That's clearly a mistake :)

I introduced it when working on #118, but I got things wrong. The polling within [`useResourceDelayedStatus`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/b919ee59566392bad320630c250238ffdd693bae/src/app/components/SEStatusLabel/useResourceDelayedStatus.ts) was not cleared at all. Fixed now.

Thanks @manstis for bringing this up.

Reply:
Thank-you for the detailed explanation.